src/Lecture2.hs

```diff
@@ -62,7 +72,7 @@ lazyProduct = error "TODO"
 "ccaabb"
 -}
 duplicate :: [a] -> [a]
-duplicate = error "TODO"
+duplicate list = concatMap (replicate 2) list 
```

Comment:
That's a correct solution 👍🏻 However, the existing implementation allocates temporary lists inside and appends them with `++`.

It is more efficient to construct lists using the cons operator (`:`) directly without introducing intermediate values or adding lists with `(++)` whenever possible.